integration/src/test/scala/com/snowplowanalytics/snowplow/collectors/scalastream/intergation/Containers.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2013-2022 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0, and
+ * you may not use this file except in compliance with the Apache License
+ * Version 2.0.  You may obtain a copy of the Apache License Version 2.0 at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the Apache License Version 2.0 is distributed on an "AS
+ * IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the Apache License Version 2.0 for the specific language
+ * governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.collectors.scalastream.intergation
+
+import cats.effect.{Resource, Sync}
+import com.dimafeng.testcontainers.GenericContainer
+import org.slf4j.LoggerFactory
+import org.testcontainers.containers.output.Slf4jLogConsumer
+import org.testcontainers.containers.wait.strategy.Wait
+import org.testcontainers.containers.{BindMode, Network, GenericContainer => JGenericContainer}
+import org.testcontainers.images.builder.ImageFromDockerfile
+
+object Containers {
+  private val network = Network.newNetwork()
+
+  def localstack: JGenericContainer[_] = {
+    val container = GenericContainer(
+      dockerImage  = "localstack/localstack-light:1.2.0",
+      exposedPorts = Seq(4566, 4567, 4568),
+      env          = Map("SERVICES" -> "kinesis", "DEFAULT_REGION" -> "eu-central-1", "USE_SSL" -> "1"),
+      waitStrategy = Wait.forLogMessage(".*AWS kinesis.CreateStream.*", 2),
+      fileSystemBind = Seq(
+        GenericContainer.FileSystemBind(
+          "./.localstack",
+          "/var/lib/localstack",
+          BindMode.READ_WRITE
+        ),
+        GenericContainer.FileSystemBind(
+          "integration/src/test/resources/localstack",
+          "/docker-entrypoint-initaws.d",
+          BindMode.READ_ONLY
+        ),
+        GenericContainer.FileSystemBind(
+          "/var/run/docker.sock",
+          "/var/run/docker.sock",
+          BindMode.READ_WRITE
+        )
+      )
+    )
+    container.underlyingUnsafeContainer.withNetwork(network)
+    container.underlyingUnsafeContainer.withNetworkAliases("localstack")
+    container.container
+  }
+
+  def collector(flavour: String, configName: String, dep: Option[JGenericContainer[_]] = None): JGenericContainer[_] = {
+    val imageFromDockerfile = new ImageFromDockerfile()
+      .withDockerfile(java.nio.file.Path.of(flavour, "target", "docker", "stage", "Dockerfile"))
+    val container = GenericContainer(
+      dockerImage  = imageFromDockerfile,
```

Comment:
Initially, this depended on an image published with `Docker / publishLocal`. However, this means that we can only have tests for either the `focal` or the `distroless` flavour of the image, because only one of them can be tagged `latest` at publish time. This means that we can't easily specify an image to use. We need to figure out what the latest concrete tag passed to `publishLocal` is. While not necessarily impossible, I think depending on some obscure value from a third-party plugin like `dynVer` or `BuildInfo` is not going to make things much clearer.

I agree it would be neater if we can obtain the path to the staged Dockerfile programmatically. I haven't looked into it really, but it's in line with your other comments about limiting the use of magic numbers.

Reply:
Also, I realised that `sbt 'project kinesisDistroless' 'Docker / stage'` writes to a path that has a different structure, so when adding tests for the distroless flavours, this part will need some careful consideration about how to do it best.

Reply:
> this means that we can only have tests for either the focal or the distroless flavour of the image

That's fine, we only care about `distroless` image. TBH I wonder why we still publish the other one. I think that we'll stop at some point.

> only one of them can be tagged latest at publish time.

`distroless` version is [published](https://github.com/snowplow/stream-collector/blob/master/.github/workflows/deploy.yml#L106) with `latest-distroless` tag.

But we do that only for a final release, so we can't use that in the tests. I think that we have to get the version dynamically from the code. This should be possible. I also need to do that for `collector-pubsub` test, I'll let you know how if I manage.

Reply:
I've changed it so that we get the current version from `BuildInfo` in the source code.